cmd/installer/configureStorage.go

```diff
@@ -52,8 +59,94 @@ func init() {
 	gob.Register(&filesystem.DirectoryInode{})
 }
 
+func configureBootDrive(cpuSharer cpusharer.CpuSharer, drive *driveType,
+	layout installer_proto.StorageLayout, bootPartition int, img *image.Image,
+	objGetter objectserver.ObjectsGetter, logger log.DebugLogger) error {
+	startTime := time.Now()
+	if run("blkdiscard", *tmpRoot, logger, drive.devpath) == nil {
+		drive.discarded = true
+		logger.Printf("discarded %s in %s\n",
+			drive.devpath, format.Duration(time.Since(startTime)))
+	} else {
+		if err := eraseStart(drive.devpath, logger); err != nil {
+			return err
+		}
+	}
+	args := []string{"-s", "-a", "cylinder", drive.devpath, "mklabel", "msdos"}
+	unitSize := uint64(1 << 20)
+	unitSuffix := "MiB"
+	offsetInUnits := uint64(1)
+	for _, partition := range layout.BootDriveLayout {
+		sizeInUnits := partition.MinimumFreeBytes / unitSize
+		if sizeInUnits*unitSize < partition.MinimumFreeBytes {
+			sizeInUnits++
+		}
+		args = append(args, "mkpart", "primary", "ext2",
+			strconv.FormatUint(offsetInUnits, 10)+unitSuffix,
+			strconv.FormatUint(offsetInUnits+sizeInUnits, 10)+unitSuffix)
+		offsetInUnits += sizeInUnits
+	}
+	args = append(args, "mkpart", "primary", "ext2",
+		strconv.FormatUint(offsetInUnits, 10)+unitSuffix, "100%")
+	args = append(args,
+		"set", strconv.FormatInt(int64(bootPartition), 10), "boot", "on")
+	if err := run("parted", *tmpRoot, logger, args...); err != nil {
+		return err
+	}
+	// Prepare all file-systems concurrently, make them serially.
+	concurrentState := concurrent.NewState(uint(
```

Comment:
Concurrent? this is new lib? what does it get you?

Reply:
It makes it easier to run goroutines and reap (and stop) the error code. It's been around since Feb-2016.